Lib/defconQt/tools/drawing.py

```diff
@@ -225,6 +230,59 @@ def drawFontVerticalMetrics(painter, glyph, scale, rect, drawLines=True,
             drawTextAtPoint(painter, text, x, y, scale)
     painter.restore()
 
+# Guidelines
+
+def drawFontGuidelines(painter, glyph, scale, rect, drawLines=True,
+                       drawText=True):
+    """
+    Draws the font guidelines of the Glyph_ *glyph* in the form of lines if
+    *drawLines* is true and text if *drawText* is true using QPainter_
+    *painter*.
+
+    *rect* specifies the rectangle which the lines will be drawn in (usually,
+    that of the glyph’s advance width).
+
+    .. _Glyph: http://ts-defcon.readthedocs.org/en/ufo3/objects/glyph.html
+    .. _QPainter: http://doc.qt.io/qt-5/qpainter.html
+    """
+    if not drawLines and not drawText:
+        return
+    font = glyph.font
+    if font is None:
+        return
+    xMin = rect[0]
+    xMax = xMin + rect[2]
+    yMin = rect[1]
+    yMax = yMin + rect[3]
+    fontSize = 9
+    for line in font.info.guidelines:
+        x1 = xMin
+        x2 = xMax
+        y1 = yMin
+        y2 = yMax
+        if line.y is not None:
+            y1 = y2 = line.y
+        if line.x is not None:
+            x1 = x2 = line.x
+        color = defaultColor("fontGuideline")
+        if line.color:
+            color = colorToQColor(line.color)
+        painter.save()
+        painter.setPen(color)
+        if drawLines:
+            drawLine(painter, x1, y1, x2, y2)
```

Comment:
Does this actually draw a line when "x" and "y" are set on the guideline? 
Because x1 = x2 = line.x and y1 = y2 = line.y will make a zero-length line, i.e. draw nothing. Spec says:

> The guideline extends along angle to infinity in both directions out of the point defined by x and y.